### PR TITLE
K8SPSMDB-365: Allow backups even if just a single node is available

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPSMDB-365](https://jira.percona.com/browse/K8SPSMDB-365) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/668/commits/8c20013c5241120f941642a9147ea6bf044e4810
image: `perconalab/percona-server-mongodb-operator:PR-668-8c20013c`

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| pitr-sharded | passed |
| users | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/668/commits/42a11e9aa6776ed633210646d85016d09a4be609
image: `perconalab/percona-server-mongodb-operator:PR-668-42a11e9a`

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 1]
```diff
@@ -604,3 +606,21 @@ func (cr *PerconaServerMongoDB) StatefulsetNamespacedName(rsName string) types.N
 func (cr *PerconaServerMongoDB) MongosNamespacedName() types.NamespacedName {
 	return types.NamespacedName{Name: cr.Name + "-" + "mongos", Namespace: cr.Namespace}
 }
+
+func (cr *PerconaServerMongoDB) CanBackup() error {
+	if cr.Status.State == AppStateReady {
+		return nil
+	}
+
+	if !cr.Spec.UnsafeConf {
```
Comment (qJkee) [inline pkg/apis/psmdb/v1/psmdb_types.go:615]: @delgod @hors 
Should we do this?
In that case, user will have to set this value to true, which could cause other issues with cluster

**Review by qJkee — CHANGES_REQUESTED** [review]

File: pkg/apis/psmdb/v1/psmdb_types.go (hunk 1 above)
Comment (egegunes) [reply pkg/apis/psmdb/v1/psmdb_types.go:615]: It's based on the comments in the issue: https://jira.percona.com/browse/K8SPSMDB-365?focusedCommentId=278051&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-278051

Comment (delgod) [reply pkg/apis/psmdb/v1/psmdb_types.go:615]: this thing is escalated by Support, please read the initial issue

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 2]
```diff
@@ -604,3 +606,21 @@ func (cr *PerconaServerMongoDB) StatefulsetNamespacedName(rsName string) types.N
...
+		return errors.New(fmt.Sprintf("allowUnsafeConfigurations must be true to run backup on cluster with status %s", cr.Status.State))
```
[the remark this answers is not in the document]
Comment (egegunes) [reply pkg/apis/psmdb/v1/psmdb_types.go]: fixed

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| rs-shard-migration | failed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/668/commits/6a541d87be35f0cda9739f112193f3a9d048e1ae
image: `perconalab/percona-server-mongodb-operator:PR-668-6a541d87`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| arbiter | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| version-service | passed |
| pitr | passed |
| users | passed |
| pitr-sharded | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/668/commits/6a541d87be35f0cda9739f112193f3a9d048e1ae
image: `perconalab/percona-server-mongodb-operator:PR-668-6a541d87`

**Review by qJkee — APPROVED** [review]

**Pull request merged**